lib/krane/kubernetes_resource.rb

```diff
@@ -350,13 +355,20 @@ def sensitive_template_content?
     end
 
     def server_dry_runnable_resource?
-      self.class::SERVER_DRY_RUNNABLE
+      self.class::SERVER_DRY_RUNNABLE && @definition.dig('metadata', 'generateName').blank?
```

Comment:
```suggestion
      # generateName and server-side dry run are incompatible because the former only works with `create` and the latter only works with `apply`
      self.class::SERVER_DRY_RUNNABLE && @definition.dig('metadata', 'generateName').blank?
```

Reply:
Sidenote: if you make a `uses_generate_name?` method, this (and maybe other places?) could also use it